lib/datadog/tracing/contrib/elasticsearch/patcher.rb

```diff
@@ -31,14 +31,26 @@ def patch
 
           # rubocop:disable Metrics/MethodLength
           # rubocop:disable Metrics/AbcSize
+          # rubocop:disable Metrics/CyclomaticComplexity
+          # rubocop:disable Metrics/PerceivedComplexity
           def patch_elasticsearch_transport_client
             # rubocop:disable Metrics/BlockLength
-            ::Elasticsearch::Transport::Client.class_eval do
+            transport_module::Client.class_eval do
               alias_method :perform_request_without_datadog, :perform_request
               remove_method :perform_request
 
               def perform_request(*args)
-                service = Datadog.configuration_for(self, :service_name) || datadog_configuration[:service_name]
+                # DEV-2.0: Remove this access, as `Client#self` in this context is not exposed to the user
+                # since `elasticsearch` v8.0.0. In contrast, `Client#transport` is always available across
+                # all `elasticsearch` gem versions and should be used instead.
+                service = Datadog.configuration_for(self, :service_name)
```

Comment:
Great idea, added!

Reply:
Minor minor: I wonder if we should reference/point to https://github.com/DataDog/dd-trace-rb/blob/master/docs/Deprecation.md in the deprecation message.

Reply:
I was thinking of having an explicit `Datadog.logger.deprecation_warning` (or `Datadog::Core.deprecation_warning`, not sure exactly where), so we have consistent messaging around these messages.
I think linking to the deprecation doc you mentioned is a good idea.

Reply:
❤️ I really like the idea of a shared deprecation helper -- we get a more consistent experience rather than every warning being a slightly different variant which may or may not include all the info we need.